doc/migration_guide.rst

```diff
@@ -41,19 +48,118 @@ Functionality removed from the TLS implementation includes
 * DHE_PSK ciphersuites
 * CECPQ1 ciphersuites
 
-TLS 1.3 API adaptions
----------------------
+enum classes
+^^^^^^^^^^^^
+
+The publicly available C++ enums in the TLS namespace are now `enum class` and
+their member naming scheme was converted from `SHOUTING_SNAKE_CASE` to
+`CamelCase`.
+
+Callbacks
+^^^^^^^^^
+
+A number of new callbacks were added with TLS 1.3. None of those new callbacks
+is mandatory to implement by applications, though. Additionally there are a few
+backward incompatible changes in callbacks that might require attention by some
+applications:
+
+tls_verify_cert_chain()
+"""""""""""""""""""""""
+
+The parameter `ocsp_responses` is now a vector of `std::optional<OCSP::Response>`
+rather than potentially null `std::shared_ptr<OCSP::Response>`.
+
+tls_modify_extensions() / tls_examine_extensions()
+""""""""""""""""""""""""""""""""""""""""""""""""""
+
+These callbacks now have an additional parameter of type `Handshake_Type` that
+identify the TLS handshake message the extensions in question are residing in.
+TLS 1.3 makes much heavier use of such extensions in a wider range of messages
+to implement core protocol functionality.
 
-Sessions
-^^^^^^^^
+Policy
+^^^^^^
+
+choose_key_exchange_group()
+"""""""""""""""""""""""""""
+
+The new parameter `offered_by_peer` identifies the key exchange groups a peer
+has sent public exchange offerings for (in TLS 1.3 handshakes only).
```

Comment:
Maybe include something like "In TLS 1.2, this vector is always empty" just so it's clear that this function is still called in TLS 1.2 (IIUC), just missing this information.